packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -93,6 +99,14 @@ export default function createCssVarsProvider(options) {
       storageWindow,
     });
 
+    let mode = stateMode;
+    let colorScheme = stateColorScheme;
+
+    if (nested) {
```

Comment:
Instead of coming up with new prop `disableNestedContext`, can we simply inherit them from the context if they exist, and override from the actual props in the `CssVarsProvider` if they are explicitly specified. This is usually how it works with all contexts, you inherit from the parent, and if someone is specified in the nested context, it wins.

Reply:
But the `stateMode` and `stateColorScheme` comes from the internal hook including the state updaters. If my understanding is correct, you meant the `mode` and `colorScheme` could be controlled from the outside?

Reply:
If yes, developers have to build their own version of the state or we have to export the internal hook. I don't think that will be a better experience than using `disableNestedContext`.

Reply:
Right, I forgot we have a hook for it. Thanks for the clarificaiton.